### PR TITLE
feat: allow error layouts to be added through addLayout

### LINKED ISSUE #3194: this.addLayout of module not possible to set error layout
### What problem does this feature solve?

Currently only `layouts/error.vue` is able to overwrite the error layout. I'd like to do this inside of a module with:
```js
this.addLayout({
 name:"error",
 src: "path/to/custom/error.vue"
})
```
It would be helpful for module to developer to overwrite the error page on demand.

<div align="right"><sub><em>This feature request is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c6850">#c6850</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by TheAlexLichter**
Resolves: https://github.com/nuxt/nuxt.js/issues/3194

### CODE REVIEW

**Review by galvez — APPROVED** [review]
LGTM

File: lib/core/module.js [hunk 1]
```diff
@@ -71,6 +71,12 @@ export default class ModuleContainer {
 
     // Add to nuxt layouts
     this.options.layouts[name || path.parse(src).name] = `./${dst}`
+
+    // If error layout, set ErrorPage
+    if (name === 'error') {
```
Comment (pi0) [inline lib/core/module.js]: Can we add it as another function like `setErrorLayout`? This is somehow a hidden behaviour

Comment (manniL) [reply lib/core/module.js]: Done :relaxed:

**Review by pi0 — APPROVED** [review]

**Comment by manniL** [comment]
@willin Thanks for you feedback. In general the docs for the module functions (like `addLayout`) are missing. We will add them shortly.

**Pull request merged**